### ISSUE TITLE
TypeError: dir() takes exactly 1 arguments (0 given) 

### ISSUE THREAD

**Issue by QGB** [skulpt/skulpt#758]
in cpython
Docstring:
dir([object]) -> list of strings

If called without an argument, return the names in the current scope.
Else, return an alphabetized list of names comprising (some of) the attributes
of the given object, and of attributes reachable from it.
If the object supplies a method named __dir__, it will be used; otherwise
the default dir() logic is used and returns:
  for a module object: the module's attributes.
  for a class object:  its attributes, and recursively the attributes
    of its bases.
  for any other object: its attributes, its class's attributes, and
    recursively the attributes of its class's base classes.
Type:      builtin_function_or_method

**Comment by QGB** [comment]
NotImplementedError: locals is not yet implemented on line 7
NotImplementedError: vars is not yet implemented on line 8

globals() implemented

**Comment by albertjan** [comment]
yeah, this is one of the big oustanding pain points for skulpt, we can't have a `dir` without args because `locals` and `globals` don't work correctly. I've attempted to fix this a few times but it's very difficult to emulate how python deals with variable scoping. See: #16
